File: app/src/main/java/ru/fmproject/android/calculator/MyExceptions.java
package ru.fmproject.android.calculator;

public class MyExceptions extends Throwable {

    public static final int NONE = 0;
    public static final int DIVIDE_BY_ZERO = 1;
    public static final int NOT_INTEGER = 2;
    public static final int TOO_BIG = 3;
    public static final int TOO_SMALL = 4;
    public static final int NEGATIVE = 5;
    public static final int INFINITY = 6;
    public static final int NEGATIVE_INFINITY = 7;
    public static final int NOT_NUMBER = 8;

    private final String[] exp = {
            "",
            "Деление на ноль.",
            "Ожидается целое число.",
            "Очень большое число.",
            "Очень маленькое число.",
            "Ожидается положительное число.",
            "Бесконечность.",
            "Минус бесконечность.",
            "Ожидается число.",
            "Запрос был перенаправлен по другому адресу."
    };

    private final int reason;
    private final String msg;


    public MyExceptions() {
        this(NONE, "");
    }

    public MyExceptions(int reason) {
        this(reason, "");
    }

    public MyExceptions(int reason, String msg) {
        this.reason = reason;
        this.msg = msg;

    }

    public String getReason() {
        return exp[reason];
    }

    public String getMsg() {
        return msg;
    }
}

File: app/src/main/java/ru/fmproject/android/calculator/MyExceptions.kt
package ru.fmproject.android.calculator

/**
 * Created by User on 23.06.2017.
 */

internal class MyExceptions : Throwable {

    var exp = arrayOf("", "Деление на ноль.", "Ожидается целое число.", "Очень большое число.", "Очень маленькое число.", "Ожидается положительное число.", "Бесконечность.", "Минус бесконечность.", "Ожидается число.", "Запрос был перенаправлен по другому адресу.")

    var msg = ""
    private var reason: Int = 0


    constructor(reason: Int) {
        this.reason = reason
    }

    constructor(reason: Int, msg: String) {
        this.reason = reason
        this.msg = msg
    }

    fun getReason(): String {
        return exp[reason]
    }

    companion object {

        val NONE = 0
        val DIVIDE_BY_ZERO = 1
        val NOT_INTEGER = 2
        val TOO_BIG = 3
        val TOO_SMALL = 4
        val NEGATIVE = 5
        val INFINITY = 6
        val NEGATIVE_INFINITY = 7
        val NOT_NUMBER = 8
        val REDIRECT = 9
    }
}
